activemq-unit-tests/src/test/java/org/apache/activemq/usecases/MaxUncommittedCountExceededTest.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.broker.BrokerService;
+import org.apache.activemq.broker.jmx.BrokerMBeanSupport;
+import org.apache.activemq.broker.jmx.QueueViewMBean;
+import org.apache.activemq.command.ActiveMQDestination;
+import org.apache.activemq.command.ActiveMQQueue;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+import jakarta.jms.Connection;
+import jakarta.jms.JMSException;
+import jakarta.jms.Message;
+import jakarta.jms.MessageProducer;
+import jakarta.jms.Queue;
+import jakarta.jms.ResourceAllocationException;
+import jakarta.jms.Session;
+
+public class MaxUncommittedCountExceededTest {
+
+    public static final String DEFAULT_JMX_DOMAIN_NAME = "org.apache.activemq";
+    public static final String DEFAULT_JMX_BROKER_NAME = "localhost";
+
+    public static final String DEFAULT_JMS_USER = "admin";
+    public static final String DEFAULT_JMS_PASS = "admin";
+
+    protected ActiveMQConnectionFactory activemqConnectionFactory = null;
+    protected BrokerService brokerService = null;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    // Control session
+    protected Connection connection = null;
+    protected Session session = null;
+    protected MessageProducer messageProducer = null;
+
+    protected String methodNameDestinationName = null;
+    protected MBeanServer mbeanServer = null;
+    protected QueueViewMBean queueViewMBean = null;
+
+    @Before
+    public void setUp() throws Exception {
+        brokerService = new BrokerService();
+        brokerService.setDeleteAllMessagesOnStartup(true);
+        brokerService.setPersistent(true);
+        brokerService.setUseJmx(true);
+        brokerService.addConnector("tcp://localhost:0").setName("Default");
+        brokerService.setBrokerName("localhost");
+        brokerService.start();
+        brokerService.waitUntilStarted(30_000);
+        brokerService.deleteAllMessages();
+        assertNotNull(brokerService);
+
+        activemqConnectionFactory = new ActiveMQConnectionFactory(brokerService.getTransportConnectorByScheme("tcp").getPublishableConnectString());
+        connection = activemqConnectionFactory.createConnection();
+        connection.start();
+        session = connection.createSession(true, Session.SESSION_TRANSACTED);
+        methodNameDestinationName = "AMQ.TX." + cleanParameterizedMethodName(testName.getMethodName().toUpperCase());
+        Queue queue = session.createQueue(methodNameDestinationName);
+        messageProducer = session.createProducer(queue);
+        mbeanServer = ManagementFactory.getPlatformMBeanServer();
+        brokerService.getAdminView().addQueue(methodNameDestinationName);
+        queueViewMBean = getQueueViewMBean(new ActiveMQQueue(methodNameDestinationName));
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (messageProducer != null) {
```

Comment:
The code here to close the producer and session seems un-necessary as closing the connection will also close these. You also don't really need to set the reference to null or any references to null as they will be reset in the setUp() method when running in the future or just cleaned up by the garbage collector when the test is done. 

It's not a huge deal but the code base has a ton of this kind of extra code laying around that is unnecessary and just leads to more bloat. We really should just create utilities or  move this clean up stuff to a common section anyways.

Reply:
Agreed, I'd like to build a JUnit 5 set of tools to provide all this boilerplate code.

Reply:
Fixed.